spec/src/antlr/JDQL.g4

```diff
@@ -0,0 +1,93 @@
+grammar JDQL;
+
+statement : select_statement | update_statement | delete_statement;
+
+select_statement : select_clause? from_clause? where_clause? orderby_clause?;
+update_statement : 'UPDATE' entity_name set_clause where_clause?;
+delete_statement : 'DELETE' from_clause where_clause?;
+
+from_clause : 'FROM' entity_name;
+
+where_clause : 'WHERE' conditional_expression;
+
+set_clause : 'SET' update_item (',' update_item)*;
+update_item : state_field_path_expression '=' (scalar_expression | 'NULL');
+
+select_clause : 'SELECT' select_list;
+select_list
+    : state_field_path_expression
+    | aggregate_expression
+    ;
+aggregate_expression : 'COUNT' '(' '*' ')';
+
+orderby_clause : 'ORDER' 'BY' orderby_item (',' orderby_item)*;
+orderby_item : state_field_path_expression ('ASC' | 'DESC');
+
+conditional_expression
```

Comment:
JPQL has empty_collection_comparison_expression and I think Query by Method Name has an `Empty` keyword, so we should probably include it in order to be able to transform from Query by Method Name to query language.

Similar for collection_member_expression and Collection `Contains`.

Reply:
Yeah and Query by Method Name also currently has something similar to JPQL's `member of`. 

I mentioned this earlier.

The problem is that currently we don't have any sort of entity field that can be a collection, so that functionality just isn't well-defined. There's nothing for `empty` or `member of` to operate on!

Either:

1. we need to define some kind of entity field that can be a collection, or
2. simply remove that feature from Query by Method Name as well.

Reply:
Yes, the NoSQL often includes support for multi-valued fields or embeddables, allowing for complex data types and structures directly within documents or records.

Reply:
Of course, but I was more asking what the Jakarta NoSQL _specification_ actually says about this.

Reply:
Whether to have query language for collection attributes can be decided in a separate pull.  My intent here was only to ensure we consider what we want here.

Reply:
Yup. Should we open an issue?

Reply:
Yes, let's open a separate issue.

Reply:
But in my option, I would go to simple remove that feature from query by method name as well.
The second option.

Reply:
As discussed today, 43c88751ec5adf885c1d7343ea42703b2d475aff removes `Contains` and `Empty` from QbMN, which alleviates the need for `empty` and `member of`.